### PR TITLE
Fix logout cancelling

### LINKED ISSUE #6401: Logout/Reboot Canceled by Clementine in KDE5
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [ x] I checked the issue tracker for similar issues
- [ x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [ x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System:
Arch Linux with KDE Plasma
- Clementine version: 
1.3.1.r771.75f18dab2-1

### Expected behaviour / actual behaviour
I just click reboot and even with music stopped, if clementine is in the background it prevents me from rebooting. Just annoying

### Steps to reproduce the problem (only for bugs)
Only happens in KDE5 as far as I know.
1.) Open clementine
2.) Click reboot/logout
3.) The reboot process is stopped with a popup saying clementine is responsible

### PR DESCRIPTION
**Pull request by Korvox**
Closes #6401

Never heard anyone on Win / OSX having a problem with this so might as well merge it finally.

### CODE REVIEW

**Comment by jonaski** [comment]
This minimizes Clementine to the system tray instead of exiting when exiting with [X] even when the keep running setting is off.
You just removed the entire code, how does that fix anything?

**Comment by Korvox** [comment]
It fixes the bug by not ignoring close events when the tray is active. Before system close events were being ignored and thus Clementine was preventing logout.

File: src/ui/mainwindow.cpp [hunk 1]
```diff
@@ -1465,17 +1465,9 @@ void MainWindow::StopAfterCurrent() {
...
-    keep_running =
-        settings_.value("keeprunning", tray_icon_->IsVisible()).toBool();
-
-  if (keep_running && event->spontaneous()) {
-    event->ignore();
-    SetHiddenInTray(true);
-  } else {
-    Exit();
-  }
+  if (!tray_icon_ || !settings_.value("keeprunning", tray_icon_->IsVisible()).toBool())
```
Comment (jonaski) [inline src/ui/mainwindow.cpp:1468]: This will now crash on systems without a system tray icon.

Comment (Korvox) [reply src/ui/mainwindow.cpp:1468]: No it won't, if tray_icon_ is null it short circuits to Exit.

Unless there is some circumstance where tray_icon_ is non-null but tray_icon_->IsVisible is an invalid op? But that would have crashed the previous code.

Comment (jonaski) [reply src/ui/mainwindow.cpp:1468]: Yes your right, my bad

**Comment by hatstand** [comment]
If you rebase, CI might pass.

**Pull request merged**